src/contract.rs

```diff
@@ -0,0 +1,828 @@
+                amount: vec![Coin {
+                    denom: deal.bid_token_denom.clone(),
+                    amount: bid.amount,
+            }]};
+            refund_messages.push(refund_to_bidder);
+         }
+    }
+
+    //transfering of bid_token_denoms to deal creator
+    let transfer_to_dealer: BankMsg = BankMsg::Send{
+        to_address: deal.deal_creator.to_string().clone(),
+        amount: vec![Coin {
+            denom: deal.bid_token_denom.clone(),
+            amount:deal.total_bid.min(deal.deal_token_amount),
+    }]};
+    deal_transfer_messages.push(transfer_to_dealer);
+    if remaining_deposit > Uint128::zero(){
+          //refunding to deal_creator
+          let refund_to_deal_creator: BankMsg = BankMsg::Send{
+            to_address:deal.deal_creator.to_string().clone(),
+            amount: vec![Coin {
+                denom: deal.deal_token_denom.clone(),
+                amount:remaining_deposit,
+        }]};
+        refund_messages.push(refund_to_deal_creator);
+    }
+    let all_messages = refund_messages
+    .into_iter()
+    .chain(bid_transfer_messages.into_iter())
+    .chain(deal_transfer_messages.into_iter())
+    .collect::<Vec<_>>();
+    Ok(Response::new().add_messages(all_messages).add_attribute("action", "deal executed"))
+}
+
+
+pub fn execute_deal(deps: DepsMut, _env: Env, _info: MessageInfo, msg: ExecuteDealMsg) -> Result<Response,ContractError> {
+    // Logic to execute the deal if conditions are met
+    let query_deal=  DEALS.load(deps.storage,msg.deal_id);
+    let deal = match query_deal {
+        Ok(deal) => deal,
+        Err(_err) => return Err( ContractError::DealNotExisted {}),
+    };
+
+    let end_block_height=Uint128::from(_env.block.height);
+    // if end_block_height <= deal.end_block {
+    //     println!("block height is {}",end_block_height);
+    //      return Err(ContractError::DealTimeNotFulfilled{});
+    // }
+
+    
+    let min_cap=deal.min_cap;
+   // let mut bid_store = DEALSTORE.may_load(deps.storage, msg.deal_id)?;
+        // Loading  bid store 
+        let mut bid_store = match DEALSTORE.may_load(deps.storage, msg.deal_id)? {
+            Some(store) => store,
+            None => return Err(ContractError::BidStoreNotFound {}),
+        };
+
+        // Sort bid store by price descending order and bid id's
+        sort_by_price_desc(&mut bid_store);
+        println!("Sorted BidStore: {:?}", bid_store);
+        
+        // Checking  if total bid deposit meets minimum capacity
+        if deal.total_bid < min_cap.into() {
+             let _res = refund_bids(&mut bid_store, _info.clone(), deal);
+             DEALS.remove(deps.storage,msg.deal_id);
+            return _res;
+        }
+        // Distribute tokens
+        let _resp = distribute_tokens(&mut bid_store, deal.total_bid, deal.deal_token_amount, _info.clone(), deal);
+        println!("After transferring the bid store looks like {:?}", bid_store);
```

Comment:
ditto

Reply:
Removed,it used for just log to debugging